pkg/controller/deployment/progress_test.go

```diff
@@ -176,7 +178,10 @@ func TestRequeueStuckDeployment(t *testing.T) {
 			if test.nowFn != nil {
 				nowFn = test.nowFn
 			}
-			got := dc.requeueStuckDeployment(test.d, test.status)
+			_, ctx := ktesting.NewTestContext(t)
+			ctx, cancel := context.WithCancel(ctx)
+			defer cancel()
```

Comment:
If the test doesn't spawn any goroutines, then WithCancel isn't needed. Applies also below.